build.rs

```diff
@@ -1,17 +1,27 @@
 // Copyright 2018 the Deno authors. All rights reserved. MIT license.
 
 // Run "cargo build -vv" if you want to see gn output.
-// TODO For the time being you must set an env var DENO_BUILD_PATH
-// which might be `pwd`/out/debug or `pwd`/out/release.
-// TODO Currently DENO_BUILD_PATH must be absolute.
 // TODO Combine DENO_BUILD_PATH and OUT_DIR.
 
 use std::env;
+use std::path::Path;
 use std::process::Command;
 
 fn main() {
+  // Cargo sets PROFILE to either "debug" or "release", which conveniently
+  // matches the build modes we support.
   let mode = env::var("PROFILE").unwrap();
-  let deno_build_path = env::var("DENO_BUILD_PATH").unwrap();
+
+  let repo_root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+  let repo_root_dir = Path::new(&repo_root_dir);
+
+  // TODO: use $OUT_DIR instead of "out/$PROFILE" as the default directory,
+  // once we properly skip the targets that are not needed by RLS.
```

Comment:
"once we properly skip the targets that are not needed by RLS."

what does that mean?

Reply:
1. Incremental builds are fast. 
2. cargo unfortunately doesn't tell us if `check` or `build` is being run- from what I can tell. So we can't skip ninja.

Reply:
> cargo unfortunately doesn't tell us if check or build is being run- from what I can tell. So we can't skip ninja.

I'm sure there's something in here that gives it away:

```
CARGO_CFG_DEBUG_ASSERTIONS=
CARGO_CFG_PROC_MACRO=
CARGO_CFG_TARGET_ARCH=x86_64
CARGO_CFG_TARGET_ENDIAN=little
CARGO_CFG_TARGET_ENV=msvc
CARGO_CFG_TARGET_FAMILY=windows
CARGO_CFG_TARGET_FEATURE=fxsr,sse,sse2
CARGO_CFG_TARGET_OS=windows
CARGO_CFG_TARGET_POINTER_WIDTH=64
CARGO_CFG_WINDOWS=
CARGO_HOME=C:\Users\BertBelder\.cargo
CARGO_MAKEFLAGS=--jobserver-fds=__rust_jobserver_semaphore_1024240462 --jobserver-auth=__rust_jobserver_semaphore_1024240462
CARGO_MANIFEST_DIR=d:\deno2
CARGO_PKG_AUTHORS=
CARGO_PKG_DESCRIPTION=
CARGO_PKG_HOMEPAGE=
CARGO_PKG_NAME=deno
CARGO_PKG_VERSION_MAJOR=0
CARGO_PKG_VERSION_MINOR=0
CARGO_PKG_VERSION_PATCH=0
CARGO_PKG_VERSION_PRE=
CARGO_PKG_VERSION=0.0.0
CARGO=\\?\C:\Users\BertBelder\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\rls.exe
LD_LIBRARY_PATH=C:\Users\BertBelder\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib
RUST_RECURSION_COUNT=1
RUST_SRC_PATH=C:\Users\BertBelder\.rustup\toolchains\stable-x86_64-pc-windows-msvc/lib/rustlib/src/rust/src
RUSTC=rustc
RUSTDOC=rustdoc
RUSTFLAGS=
RUSTUP_HOME=C:\Users\BertBelder\.rustup
RUSTUP_TOOLCHAIN=stable-x86_64-pc-windows-msvc
VSCODE_CLI=1
VSCODE_CWD=D:\deno2
VSCODE_HANDLES_UNCAUGHT_ERRORS=true
VSCODE_IPC_HOOK_EXTHOST=\\.\pipe\vscode-ipc-5570dc28-f150-4f53-8699-71a132190ec1-sock
VSCODE_IPC_HOOK=\\.\pipe\a94eff2d73547178695cbd1dc4460f01-1.28.2-main-sock
VSCODE_LOG_LEVEL=undefined
VSCODE_LOG_STACK=false
VSCODE_LOGS=C:\Users\BertBelder\AppData\Roaming\Code\logs\20181031T204117
VSCODE_NLS_CONFIG={"locale":"en-us","availableLanguages":{},"_languagePackSupport":true}
VSCODE_NODE_CACHED_DATA_DIR_22888=C:\Users\BertBelder\AppData\Roaming\Code\CachedData\7f3ce96ff4729c91352ae6def877e59c561f4850
VSCODE_PID=22888
```

Reply:
I've also checked - can't find anything.